docs/other-topics/extending-data-types.md

```diff
@@ -1,115 +1,174 @@
+  By default, if `toBindableValue` returns a string, this method will escape that string as a SQL string.
 
-  const PgTypes = DataTypes.postgres;
+```typescript
+import { DataTypes, StringifyOptions } from '@sequelize/core';
 
-  // Mandatory: map postgres datatype name
-  DataTypes.SOMETYPE.types.postgres = ['pg_new_type'];
+export class MyDateType extends DataTypes.ABSTRACT<Date> {
+  // [...] truncated example
+  
+  parseDatabaseValue(value: unknown): Date {
+    assert(typeof value === 'string', 'Expected to receive a string from the database');
+    
+    return new Date(value);
+  }
 
-  // Mandatory: create a postgres-specific child datatype with its own parse
-  // method. The parser will be dynamically mapped to the OID of pg_new_type.
-  PgTypes.SOMETYPE = function SOMETYPE() {
-    if (!(this instanceof PgTypes.SOMETYPE)) {
-      return new PgTypes.SOMETYPE();
-    }
-    DataTypes.SOMETYPE.apply(this, arguments);
+  toBindableValue(value: Date): unknown {
+    return value.toISOString();
   }
-  const util = require('util'); // Built-in Node package
-  util.inherits(PgTypes.SOMETYPE, DataTypes.SOMETYPE);
+  
+  escape(value: Date, options: StringifyOptions): string {
+    return options.dialect.escapeString(value.toISOString());
+  }
+}
+```
+
+## Modifying an existing Data Type
+
+You can inherit the implementation of an existing Data Type to customize its behavior.  
+To do so, make your class extend the Data Type you wish to modify instead of `DataTypes.ABSTRACT`.
 
-  // Mandatory: create, override or reassign a postgres-specific parser
-  // PgTypes.SOMETYPE.parse = value => value;
-  PgTypes.SOMETYPE.parse = DataTypes.SOMETYPE.parse || x => x;
+Note how the following example inherits from `DataTypes.STRING` instead of `DataTypes.ABSTRACT`:
 
-  // Optional: add or override methods of the postgres-specific datatype
-  // like toSql, escape, validate, _stringify, _sanitize...
+```typescript
+import { Sequelize, DataTypes } from '@sequelize/core';
 
+export class MyStringType extends DataTypes.STRING {
+  toSql() {
+    return 'TEXT';
+  }
 }
 ```
 
-### Ranges
+Just like with custom data types, use your Data Type class instead of the type you are extending:
 
-After a new range type has been [defined in postgres](https://www.postgresql.org/docs/current/static/rangetypes.html#RANGETYPES-DEFINING), it is trivial to add it to Sequelize.
+```typescript
+import { MyStringType } from './custom-types.js';
 
-In this example the name of the postgres range type is `SOMETYPE_range` and the name of the underlying postgres datatype is `pg_new_type`. The key of `subtypes` and `castTypes` is the key of the Sequelize datatype `DataTypes.SOMETYPE.key`, in lower case.
+const sequelize = new Sequelize('sqlite::memory:');
 
-```js
-function createTheNewDataType() {
-  // [...]
+const User = sequelize.define('User', {
+  firstName: {
+    // highlight-next-line
```

Comment:
We can fix it in a separate PR, but just noticed this on light mode;
![image](https://user-images.githubusercontent.com/13023439/200196525-a10f7814-2833-49da-84c1-8addbba82fed.png)
![image](https://user-images.githubusercontent.com/13023439/200196532-966695f7-18c2-4661-81db-4606d9ecd38a.png)

Reply:
Let's fix this in the docs rewrite PR